.github/workflows/build.yml

```diff
@@ -11,21 +11,43 @@ jobs:
     runs-on: ubuntu-latest
 
     steps:
-    - uses: actions/checkout@v4
-    - name: Setup cmake
-      uses: jwlawson/actions-setup-cmake@v2
-      with:
-        cmake-version: '3.22.x'
-    - name: Configure CMake
-      run: cmake -B cmake-build-posix -S executables/referenceApp -DCMAKE_BUILD_TYPE=${{env.BUILD_TYPE}}
-
-    - name: Build
-      run: cmake --build cmake-build-posix --target app.referenceApp -j --config ${{env.BUILD_TYPE}}
-
-    - name: Test
-      # Execute tests defined by the CMake configuration.
-      run: |
-          cmake -B cmake-build-unit-tests -S executables/unitTest -DBUILD_UNIT_TESTS=ON
-          cmake --build cmake-build-unit-tests -j4
-          ctest --test-dir cmake-build-unit-tests -j4
+      - uses: actions/checkout@v4
+      
+      # Setup CMake
+      - name: Setup CMake
+        uses: jwlawson/actions-setup-cmake@v2
+        with:
+          cmake-version: '3.22.x'
+      
+      - name: Cache Build Results
+        uses: actions/cache@v3
+        id: cache
+        with:
+          path: cmake-build-posix
+          key: ${{ github.ref }}-cmake-build-posix
 
+      - name: Configure CMake
+        if: steps.cache.outputs.cache-hit != 'true'
+        run: cmake -B cmake-build-posix -S executables/referenceApp -DCMAKE_BUILD_TYPE=${{ env.BUILD_TYPE }}
+
```

Comment:
one question: where do we define the BUILD_TYPE?

Reply:
On line number 7 in build.yml  we define the build BUILD_TYPE for the entire workflow